### PR TITLE
Fix #72 implem response xml

### LINKED ISSUE #72: Unimplemented error thrown from .responseXML precludes workarounds
Suppose we have the following in `onreadystatechange`:
```js
let result = this.responseXML || this.responseText || this.response;
```
As `responseXML` is not implemented, xhr-mock outputs the message `This feature hasn't been implmented yet. Please submit an Issue or Pull Request on Github.`

However, in the snippet above we have a workaround, as long as we're not expecting an XML `Document` in our tests.  Rather than just returning `undefined` and making the code under test try `responseText`, the mock just gives up.

### PR DESCRIPTION
**Pull request by K0d1Lu**
Have had to install core-js to make the project running, can add it to the pr if needed.
I couldn't test the error status as MockError crashes in constructor when I try to test it.

### CODE REVIEW

**Comment by K0d1Lu** [comment]
@jameslnewell how can I add you as a reviewer ?

**Comment by jameslnewell** [comment]
Oh, thanks for the PR! Apologies for the delayed response (my current employer uses GH which makes it hard to keep track of notifications in my OS repos). Will review in the next week.

**Comment by K0d1Lu** [comment]
> Oh, thanks for the PR! Apologies for the delayed response (my current employer uses GH which makes it hard to keep track of notifications in my OS repos). Will review in the next week.

Hi @jameslnewell did you have the time to check this PR ?

File: packages/xhr-mock/src/MockXMLHttpRequest.ts [hunk 1]
```diff
@@ -193,7 +195,17 @@ export default class MockXMLHttpRequest extends MockXMLHttpRequestEventTarget
   }
 
   get responseXML(): Document | null {
-    throw notImplementedError;
+    if (this.responseType === '' || this.responseType === 'document') {
+      if (
+        this.readyState === this.DONE &&
+        this.res.body() instanceof Document
```
Comment (jameslnewell) [inline packages/xhr-mock/src/MockXMLHttpRequest.ts:201]: Can we please check `Document` is defined before using it - in some environments e.g. NodeJS it won't be defined 🙏

**Review by jameslnewell — COMMENTED** [review]
👋 Thanks @K0d1Lu!

File: packages/xhr-mock/src/MockXMLHttpRequest.ts (hunk 1 above)
Comment (K0d1Lu) [reply packages/xhr-mock/src/MockXMLHttpRequest.ts:202]: Done, sorry for the delay ....

Comment (jameslnewell) [reply packages/xhr-mock/src/MockXMLHttpRequest.ts:202]: No worries, same!

**Review by jameslnewell — APPROVED** [review]

**Pull request merged**